internal/internal_event_handlers.go

```diff
@@ -728,8 +733,24 @@ func (wc *workflowEnvironmentImpl) GetVersion(changeID string, minSupported, max
 		// GetVersion for changeID is called first time (non-replay mode), generate a marker command for it.
 		// Also upsert search attributes to enable ability to search by changeVersion.
 		version = maxSupported
-		wc.commandsHelper.recordVersionMarker(changeID, version, wc.GetDataConverter())
-		_ = wc.UpsertSearchAttributes(createSearchAttributesForChangeVersion(changeID, version, wc.changeVersions))
+		changeVersionSA := createSearchAttributesForChangeVersion(changeID, version, wc.changeVersions)
+		attr, err := validateAndSerializeSearchAttributes(changeVersionSA)
+		if err == nil {
+			// Server has a limit for the max size of a single search attribute value. If we exceed the default limit
+			// do not try to upsert as it will cause the workflow to fail.
+			updateSearchAttribute := true
+			if wc.sdkFlags.tryUse(LimitChangeVersionSASize, !wc.isReplay) && len(attr.IndexedFields[TemporalChangeVersion].GetData()) >= changeVersionSearchAttrSizeLimit {
```

Comment:
Can you add an integration test that intentionally does enough versions to go over the SA size and then confirm the ones past the size are not serialized?

Reply:
Yes I already do that is the `version-loop-workflow-236.json` history.

Reply:
Test performance be damned, would you be willing to have an _integration_ test run a new workflow against a server triggering this behavior? It's mostly just to codify our expectations (e.g. server drops the max for SAs, though I don't think they ever will, this just makes us feel better).

Reply:
Yep can do